src/client/visualizations/bar-chart/improved-bar-chart/utils/x-domain.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2017-2018 Allegro.pl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Dataset, NumberRange, TimeRange } from "plywood";
```

Comment:
Again - very generic stuff

Reply:
Not really. There are few assumptions that work only for bar chart. This is only chart that can handle strings at x-axis and differs from line chart how it determines x-axis split.